server/api/activity/activity.controller.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+import Activity from './activity.model';
+var genericRepo = require("../generic/genericRepo");
+
+// Gets a single Employee from the DB
+export function show(req, res) {
+    console.log("=======activity show called======");
+    let date = new Date(),
+        y = date.getFullYear(),
+        m = date.getMonth(),
+        firstDate = new Date(y, m ,1).getTime();
+
+    return Activity.aggregate([
+        {
+            $match: {
+                employeeId: req.params.empid,
```

Comment:
What if req.params.empid is not present. We will end up in making a un-necessary DB call, we can create a validator to eliminate such condition.

Applicable to all apis which accepts input in any form.